### DISCUSSION TITLE
I need to configure allure report with my getConfigurationByFile settings

### DISCUSSION THREAD

**Discussion by sregeauto** [Shelex/cypress-allure-plugin#60] [Q&A] [answered]
/// <reference types="@shelex/cypress-allure-plugin" />
const allureWriter = require('@shelex/cypress-allure-plugin/writer');
const selectTestsWithGrep = require('cypress-select-tests/grep')
const fs = require('fs-extra')
const path = require('path')

function getConfigurationByFile (file) {
  const pathToConfigFile = path.resolve('cypress', 'config', `${file}.json`)

  return fs.readJson(pathToConfigFile)
}

module.exports = (on, config) => {
  on('file:preprocessor', selectTestsWithGrep(config))
  const file = config.env.configFile || 'cypress.dev'
  return getConfigurationByFile(file)
  
  
}
How can add below line of code to module.exports, without affecting my spec or test? 
allureWriter(on, config);
  return config;

**Comment by Shelex** [comment, marked as answer]
@sregeauto Thank you for opening a discussion!

Basically, if you already return config from "getConfigurationByFile", you can set allureWriter task just before that command:
```
module.exports = (on, config) => {
on('file:preprocessor', selectTestsWithGrep(config))
const file = config.env.configFile || 'cypress.dev'
allureWriter(on, config);
return getConfigurationByFile(file)
}
```
